vendor/bat-native-ads/src/bat/ads/internal/account/account.cc

```diff
@@ -135,20 +138,55 @@ void Account::ProcessTransactions() {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+void Account::Credit(const CreativeAdInfo& creative_ad,
+                     const AdType& ad_type,
+                     const ConfirmationType& confirmation_type) const {
+  const double value =
+      confirmation_type == ConfirmationType::kViewed ? creative_ad.value : 0.0;
+
+  transactions::Add(
+      creative_ad.creative_instance_id, value, ad_type, confirmation_type,
+      [=](const bool success, const TransactionInfo& transaction) {
+        if (!success) {
+          NotifyFailedToDepositFunds(creative_ad, ad_type, confirmation_type);
+          return;
+        }
+
+        NotifyDepositedFunds(transaction);
+
+        NotifyStatementOfAccountsDidChange();
+
+        confirmations_->Confirm(transaction);
+      });
+}
+
 void Account::TopUpUnblindedTokens() {
   if (!ShouldRewardUser()) {
     return;
   }
 
-  const WalletInfo wallet = GetWallet();
+  const WalletInfo& wallet = GetWallet();
```

Comment:
seems like `const WalletInfo wallet = GetWallet();` would be better and in the code below.